File: app/src/main/java/com/starostinvlad/fan/SerialPageScreen/SerialPageScreenContract.java
package com.starostinvlad.fan.SerialPageScreen;

import com.starostinvlad.fan.BaseMVP.MvpView;
import com.starostinvlad.fan.VideoScreen.PlayerModel.Serial;
import com.starostinvlad.fan.VideoScreen.PlayerModel.SerialPlayer;

public interface SerialPageScreenContract extends MvpView {
    void showLoading(boolean show);

    void fillPage(SerialPlayer serialPlayer);

    void openActivityWithSerial(Serial serial);

    void fillSeasonsList(Serial serial);

    void checkViewed(boolean viewed);

    void checkSubscribed(boolean subscribed);

    void fillBtn(int currentSeason, int currentEpisodeIndex);
}

File: app/src/main/java/com/starostinvlad/fan/SerialPageScreen/SerialPageScreenContract.kt
package com.starostinvlad.fan.SerialPageScreen

import com.starostinvlad.fan.BaseMVP.MvpView
import com.starostinvlad.fan.VideoScreen.PlayerModel.Serial
import com.starostinvlad.fan.VideoScreen.PlayerModel.SerialPlayer
import kotlin.Throws

interface SerialPageScreenContract : MvpView {
    fun showLoading(show: Boolean)
    fun fillPage(serialPlayer: SerialPlayer?)
    fun openActivityWithSerial(serial: Serial?)
    fun fillSeasonsList(serial: Serial)
    fun checkViewed(viewed: Boolean)
    fun checkSubscribed(subscribed: Boolean)
    fun fillBtn(currentSeasonIndex: Int, currentEpisodeIndex: Int)
}